### ISSUE TITLE
Calamari shell chain performance issue when improperly configured collators joins  

### ISSUE THREAD

**Issue by stechu** [Manta-Network/Manta#127]
TLDR: when Calamari (a pretty simple parachain runtime that is very similar to statemine) nodes that doesn't have the credential to be collator joined the network as a collator, it drags the performance and eventually halt to block production.

## Environment
* Polkadot v0.9.8: https://github.com/paritytech/polkadot/releases/tag/v0.9.8
* Cumulus [branch `Polkadot v0.9.8`]: https://github.com/paritytech/cumulus/tree/polkadot-v0.9.8 
* Rust toolchain: `rustc 1.55.0-nightly (7a16cfcff 2021-07-11)`

## How to Reproduce
1. Generate relay chain genesis json file:

```bash
../polkadot/target/release/polkadot build-spec --chain rococo-local --disable-default-bootnode --raw > rococo-local.json
```

2. Run 3 relay-chain validator nodes:

```bash
../polkadot/target/release/polkadot \
--chain rococo-local.json \
--alice \
--tmp \
--ws-port 9941 \
--unsafe-ws-external \
--unsafe-rpc-external \
--rpc-cors=all \
--validator
```
```bash
../polkadot/target/release/polkadot \
--chain rococo-local.json \
--bob \
--tmp \
--ws-port 9942 \
--unsafe-ws-external \
--unsafe-rpc-external \
--rpc-cors=all \
--validator 
```
```bash
../polkadot/target/release/polkadot \
--chain rococo-local.json \
--charlie \
--tmp \
--ws-port 9943 \
--unsafe-ws-external \
--unsafe-rpc-external \
--rpc-cors=all \
--validator 
```

3. Run 3 collator node using pre-defined dev accounts (only the first node is the valid collator). 
```bash
../cumulus/target/release/polkadot-collator \
--chain=westmint-dev \
--ws-port=9977 \
--port=31201 \
--collator \
--tmp \
--parachain-id=200 \
--alice \
--name=pc1 \
--telemetry-url 'wss://api.telemetry.manta.systems/submit/ 0' \
-- \
--chain=rococo-local.json \
--execution=wasm
```
```bash
../cumulus/target/release/polkadot-collator \
--chain=westmint-dev \
--ws-port=9988 \
--port=31202 \
--collator \
--tmp \
--parachain-id=200 \
--bob \
--name=pc2 \
--telemetry-url 'wss://api.telemetry.manta.systems/submit/ 0' \
-- \
--chain=rococo-local.json \
--execution=wasm
```
```bash
../cumulus/target/release/polkadot-collator \
--chain=westmint-dev \
--ws-port=9999 \
--port=31203 \
--collator \
--tmp \
--parachain-id=200 \
--charlie \
--name=pc3 \
--telemetry-url 'wss://api.telemetry.manta.systems/submit/ 0' \
-- \
--chain=rococo-local.json \
--execution=wasm
```

4. Manually promote chain `201` to be a parachain.

5. we can observe the block production becomes abnormal and eventually halts:

<img width="1394" alt="Screen Shot 2021-07-19 at 6 08 13 PM" src="https://user-images.githubusercontent.com/720571/126234038-b61753c3-c2f5-48cb-b205-8ba41e4d70dc.png">

<img width="1419" alt="Screen Shot 2021-07-19 at 6 09 59 PM" src="https://user-images.githubusercontent.com/720571/126234078-12622eaa-c5ff-4937-a36e-a15069d9619d.png">

## Security Implication
This seems give a cheap way to attack single collator parachain  (DoS). This issue is not deterministically reproducible on a more than 2 collator setting.
